lib/whitehall_importer/embed_body_references.rb

```diff
@@ -33,20 +33,23 @@ def embed_contacts(body, contacts)
       end
     end
 
-    def embed_images(body, images)
-      body&.gsub(/!!(\d+)/) do
-        whitehall_image_index = Regexp.last_match[1].to_i
-        image_name = images[whitehall_image_index - 1]
-        image_name.present? ? "[Image:#{image_name}]" : ""
+    def embed_files(body, files, file_type_indicator, file_type_name)
+      body&.gsub(/(\A|\n\n|\r\n\r\n|\n|\r\n)?#{file_type_indicator}(\d+)/) do
+        whitehall_new_lines = Regexp.last_match[1]
+        whitehall_file_index = Regexp.last_match[2].to_i
+
+        file_name = files[whitehall_file_index - 1]
+        new_lines = convert_new_lines(whitehall_new_lines)
+        "#{new_lines}[#{file_type_name}:#{file_name}]" if file_name.present?
       end
     end
 
-    def embed_attachments(body, attachments)
-      body&.gsub(/!@(\d+)/) do
-        whitehall_attachment_index = Regexp.last_match[1].to_i
-        attachment_name = attachments[whitehall_attachment_index - 1]
-        attachment_name.present? ? "[Attachment:#{attachment_name}]" : ""
-      end
+    def convert_new_lines(new_lines)
```

Comment:
A situation that isn't covered here is the one where the regex matches on `\A` - as that will get changed to a new line.

Though based on the revised criteria in https://github.com/alphagov/content-publisher/pull/1756#issuecomment-584568077 we should just need to be doing the doubling and can probably negate the no new line case.